### ISSUE TITLE
Support for Usb OTG, custom kernel.

### ISSUE THREAD

**Issue by sansors** [ClusterM/hakchi2#448]
Hello,

I Have look the motherboard of snes mini ( EUR version ), and just solder two resistors to connect the usb ID to the R16.

Would it be possible to add the USB OTG in the custom kernel of akchi?
This would be very convenient to extend storage space easily.

**Comment by isage** [comment]
There's actually no custom kernel used, afaik. Just a slightly patched uboot and initscripts in rootfs

**Comment by Locutus73** [comment]
If you look at this blog http://honeylab.hatenablog.jp (this guy actually made a custom kernel for NESC with USB support), it seems that the stock SNESC comes with a USB enabled kernel.

I hope this helps.

**Comment by Drakonas** [comment]
How do you propose the SNES Classic be able to provide enough power to do this?

**Comment by Locutus73** [comment]
I’m more a software than a hardware guy, but I was thinking to some sort of custom Y USB OTG cable with an added female USB micro port (or male USB A) in order to connect to the power brick for the shared ground and +5V.

**Comment by Drakonas** [comment]
Be aware this requires soldering onto the board of the SNES Classic to enable this feature. USB 2.0 cannot provide much extra power anyway, and that's what the port on the back of the SNES Classic is. A Y cable would connect, but it would split the power. The SNES Classic is already grabbing as much power as it can off the cable. I doubt it would be able to power anything else on the same line without having crashes.

**Comment by Locutus73** [comment]
Well, regarding the soldering, I think we should investigate the SNESC motherboard. 
Reading
http://honeylab.hatenablog.jp/entry/2017/02/16/171201
and
http://honeylab.hatenablog.jp/entry/2017/02/18/015102
I’m confused: I don’t understand if soldering is strictly necessary or if USB OTG mode can be forced in software (maybe kernel configuration). 
Regarding the power issue Nintendo recommends a 5V 1A (5W) USB charger and an iPad charger is rated 12W; I think the extra 7W should be enough for an USB memory stick (in order to store ROMS externally).

**Comment by Drakonas** [comment]
Oh so you want to extend the storage. I thought you wanted it for controllers. Yeah that won't be possible through USB OTG most likely. It would need to extend the memory using the onboard headers. https://github.com/ClusterM/hakchi2/issues/427 See this issue.

**Comment by Locutus73** [comment]
Why do you think USB storage through USB OTG won’t work?
According to http://honeylab.hatenablog.jp/entry/2017/02/18/015102

> There are some devices message recognized on my custom kernel.
> 
> a USB-PS Gamepad converter  will use with /dev/input/js0
> USB-Serial Converter FT232R will make /dev/ttyUSB0
> IO DATA USB Flash Driver 32GB has uses with fdisk
> PCI GU-1000Air USB-Ethernet Converter will make eth0 and ping packet to 8.8.8.8 !!

A custom kernel + Y cable seems to me a simpler solution than onboard header, considering Honeylab success reports.
I think you (I mean hakchi2 team) should contact him in order to join forces (Honeylab already uses hakchi2). IMHO an USB expanded SNESC (using just an external Y cable) would be a dream machine.

Just my two cents.

**Comment by Drakonas** [comment]
Did you not read the big fat post on the front page of that blog (How to enable NESC's microUSB connector as Host) stating that you need to connect (solder) connectors together to enable USB OTG?

**Comment by Locutus73** [comment]
Yes, but soldering two resistors doesn’t make this approach more complicated than soldering a SD adapter to the GPIO headers; on the contrary
1. Most of the kernel work seems to be done and tested by Honeylab
2. USB OTG is more generic (it allows other peripherals like controllers, network adapters, maybe multiple ones through a powered hub)
3. An external USB key can be easily detached and updated with a PC (while an internal SD leads to a cleaner looking solution)
4. Maybe a USB key is faster than a SD controlled by GPIO

Anyway I think Honeylab could help with kernel compiling in general, even for the SD<->GPIO approach. Either way I think an expandable NESC/SNESC would be a major milestone for hakchi2.
